ct-app/core/model/subgraph/providers.py

```
@@ -18,12 +19,15 @@ class ProviderError(Exception):
 
 
 class GraphQLProvider(Base):
-    def __init__(self, url: SubgraphURL):
+    def __init__(self, url: URL):
         self.url = url
         self.pwd = Path(__file__).parent.joinpath("queries")
         self._default_key = None
 
     #### PRIVATE METHODS ####
+    def _initialize_query(self, query_file: str, extra_inputs: list[str] = []):
```

Ruff

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)